src/easylink/rule.py

```diff
@@ -203,14 +270,182 @@ class InputValidationRule(Rule):
+    be split into since the incoming dataset isn't created until runtime). The
+    snakemake mechanism to handle this dynamic nature is a
+    `checkpoint <https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#data-dependent-conditional-execution/>`_
+    rule along with a directory as output.
+
+    Notes
+    -----
+    There is a known `Snakemake bug <https://github.com/snakemake/snakemake/issues/3036>`_
+    which prevents the use of multiple checkpoints in a single Snakefile. We
+    work around this by generating an empty checkpoint.txt file as part of this
+    rule. If this file does not yet exist when trying to run the :class:`AggregationRule`,
+    it means that the checkpoint has not yet been executed for the
+    particular wildcard value(s). In this case, we manually raise a Snakemake
+    ``IncompleteCheckpointException`` which Snakemake automatically handles
+    and leads to a re-evaluation after the checkpoint has successfully passed.
+    """
+
+    name: str
+    """Name of the rule."""
+    input_slots: dict[str, dict[str, str | list[str]]]
+    """This ``Implementation's`` input slot attributes."""
+    validations: list[str]
+    """Validation files from previous rule."""
+    output: list[str]
+    """Output directory path. It must be used as an input for next rule."""
+
+    def build_rule(self) -> str:
+        """Builds the Snakemake rule for this checkpoint.
+
+        Checkpoint rules are a special type of rule in Snakemake that allow for dynamic
+        generation of output files. This rule is responsible for splitting the input
+        files into chunks. Note that the output of this rule is a Snakemake ``directory``
+        object as opposed to a specific file like typical rules have.
+        """
+        # Replace the output filepath with an input_chunks subdir
+        output_dir = os.path.dirname(self.output[0]) + "/input_chunks"
+        # FIXME: handle multiple splitting
+        input_slots_to_split = self.get_input_slots_to_split(self.input_slots)
+        input_slot_to_split = input_slots_to_split[0]
+        checkpoint = f"""
+checkpoint:
+    name: "split_{self.name}_{input_slot_to_split}"
+    input: 
+        files={self.input_slots[input_slot_to_split]['filepaths']},
+        validations={self.validations},
+    output: 
+        output_dir=directory("{output_dir}"),
+        checkpoint_file=touch("{output_dir}/checkpoint.txt"),
+    params:
+        input_files=lambda wildcards, input: ",".join(input.files),
+    localrule: True
+    message: "Splitting {self.name} {input_slot_to_split} into chunks"
+    run:
+        splitter_utils.{self.input_slots[input_slot_to_split]["splitter"].__name__}(
+            input_files=list(input.files),
+            output_dir=list(output)[0],
+            desired_chunk_size_mb=0.1,
+        )"""
+        return checkpoint
+
+
+@dataclass
+class AggregationRule(Rule):
+    """A :class:`Rule` that aggregates the processed chunks of output data.
```

Comment:
It might be helpful if there was some term specifically for the output of an embarrassingly parallel step. I feel like this sentence and the next one are slightly redundant that you are just specifying that it is the output of an embarrassingly parallel step.